Milestone 1.1 : Update the documentation for End-To-End tests and Debug End-To-End wiki page. (#7)
oppia/oppia-web-developer-docs: Debug-end-to-end-tests.md

## Table of contents

* [Introduction](#introduction)

> * [How to know the version of failing test suite](#how-to-know-the-version-of-failing-test-suite)

Reviewer: This naming doesn't seem appropriate to what this file does. Can you please design a more appropriate name?

Author: Done.

Whenever you are debugging tests, you should create a debugging doc to document your work. This helps future contributors if they run into a similar bug in the future. If other people come in later to help you, they can also use the debugging doc to get up to speed on what you've already figured out. You can make a copy of [this template debugging doc](https://docs.google.com/document/d/1qRbvKjJ0A7NPVK8g6XJNISMx_6BuepoCL7F2eIfrGqM/edit?usp=sharing) to get started. Also check out the [[debugging docs wiki page|debugging-docs]].

There are many ways to go about debugging an E2E test, but here is one approach:

1. Create a [[debugging doc|debugging-docs]].
2. Look through the logs from the failed test to try and understand what went wrong. In particular:

   * Look for a line that says just `Killed`. This line indicates that some process was killed by the operating system for consuming too much memory. It's fairly safe to assume that the test failure was because of that process being killed.
   * Look for the stack trace and error message of the _first_ error. The trace might point you to where the error was thrown in the test code, and the message may help explain what went wrong.

> 3. If you don't understand what the error message means, search for it online. Also look through the test code and `core/test/webdriverior_utils/action.js` to see if the error message (or something like it) is in our test code.

Reviewer 2: For readers at this point in the docs, the problem could still be in `protractor_utils/action.js` right?

Author: Right, updated.

4. Enable video recordings and rerun the test until you reproduce the error. Then watch the video recordings and follow along in the test code. Try and understand why the error was thrown.

5. Try and reproduce the error locally. If you succeed, you can use your local debugger to investigate.

## How to know the version of failing test suite

> Currently, we have two versions of E2E test suites present: Protractor and WebDriverIO. So It's important to first find out which version of test suite is failing on your PR. The following [webdriverio migration tracker](https://docs.google.com/spreadsheets/d/1Mj-llYXMURtis54vpL2VL7BwgRiFIZ1nIFtK3fY3Se4/edit?usp=sharing) contains the information about which suite is present in which version (see Suites migrated to WebdriverIO, Suites still in Protractor)

Reviewer 2: ```suggestion
Currently, we have two versions of E2E test suites present, Protractor and WebDriverIO, so it's important to first find out which framework the failing test suite is written in. The [webdriverio migration tracker](https://docs.google.com/spreadsheets/d/1Mj-llYXMURtis54vpL2VL7BwgRiFIZ1nIFtK3fY3Se4/edit?usp=sharing) contains the information about which suite is present in which framework (see "Suites migrated to WebdriverIO" and "Suites still in Protractor").
```

Author: Done.

> 3. If you don't understand what the error message means, search for it online. Also look through the test code and `core/test/protractor_utils/action.js` to see if the error message (or something like it) is in our test code.

Reviewer 2: I think you need to mention `webdriver_utils/action.js` here too. [Before](https://github.com/oppia/oppia-web-developer-docs/pull/7#discussion_r900404545) you had replaced protractor with webdriver, but I think we need both since the error could come from either file, right?

Author: Done.
